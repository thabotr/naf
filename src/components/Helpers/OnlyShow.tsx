export function OnlyShow({
  If,
  children,
}: {
  If?: boolean;
  children: React.ReactNode;
}) {
  return If ? <>{children}</> : null;
}
