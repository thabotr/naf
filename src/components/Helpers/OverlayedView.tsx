import { View, ViewProps } from "react-native";

export function OverlayedView(props: ViewProps){
  const style: Object = props.style?.valueOf() ?? {};
  const newProps: ViewProps = {
    ...props,
    style: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      justifyContent: 'center',
      alignItems: 'center',
      ...style,
    }
  }; 
  return <View {...newProps}>
    {props.children}
  </View>
}