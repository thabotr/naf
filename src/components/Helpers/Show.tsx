export function Show({
  component,
  If,
  ElseShow,
}: {
  component: React.ReactNode;
  If: boolean;
  ElseShow: React.ReactNode;
}): JSX.Element {
  return <>{If ? component : ElseShow}</>;
}
