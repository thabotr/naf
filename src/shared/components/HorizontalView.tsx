import React from 'react';
import {View, ViewProps} from 'react-native';

export default function HorizontalView(props: ViewProps) {
  const style: Object = props.style?.valueOf() ?? {};
  const newProps: ViewProps = {
    ...props,
    style: {...style, display: 'flex', flexDirection: 'row'},
  };
  return <View {...newProps}>{props.children}</View>;
}
