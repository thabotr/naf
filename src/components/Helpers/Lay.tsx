import React from 'react';
import {OverlayedView} from './OverlayedView';

export function Lay({
  component,
  over,
}: {
  component: React.ReactNode;
  over: React.ReactNode;
}) {
  return (
    <>
      {over}
      {<OverlayedView>{component}</OverlayedView>}
    </>
  );
}
