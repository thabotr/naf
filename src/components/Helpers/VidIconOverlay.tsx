import { IconButton } from "react-native-paper";
import { OverlayedView } from "./OverlayedView";

export const vidIconOverlay= (iconSize: number, blur?:boolean) => <OverlayedView>
  <IconButton size={iconSize} icon="play-circle-outline"  style={{opacity: blur ? 0.4 : 1}}/>
</OverlayedView>