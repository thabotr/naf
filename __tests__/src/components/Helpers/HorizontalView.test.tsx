import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
import {HorizontalView} from '../../../../src/components/Helpers/HorizontalView';

test('renders child elements horizontally', () => {
  const tree = renderer
    .create(
      <HorizontalView>
        <Text>column one</Text>
        <Text>column two</Text>
      </HorizontalView>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
