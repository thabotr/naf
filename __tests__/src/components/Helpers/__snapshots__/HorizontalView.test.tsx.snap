// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders child elements horizontally 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text>
    column one
  </Text>
  <Text>
    column two
  </Text>
</View>
`;
